lib/plug/adapters/cowboy/conn.ex

```diff
@@ -37,11 +37,7 @@ defmodule Plug.Adapters.Cowboy.Conn do
   def send_file(req, status, headers, path, offset, length) do
     %File.Stat{type: :regular, size: size} = File.stat!(path)
 
-    length =
-      cond do
-        length == :all -> size
-        is_integer(length) -> length
-      end
+    length = if length == :all, do: size, else: length
```

Comment:
So the semnatics here are not the same. The previous code would error if length is not an integer or `:all`. I would keep this code as is.